### DISCUSSION TITLE
Make tailwindcss/forms configurable

### DISCUSSION THREAD

**Discussion by helhum** [tailwindlabs/tailwindcss#4649] [Ideas]
Currently there is no way to configure some of the opinionated styles provided by this plugin.
See https://github.com/tailwindlabs/tailwindcss/discussions/4645 for example.

Maybe it would be helpful to make some things configurable via the theming API like it was the case for tailwindcss/custom-forms or it is the case with tailwindcss/typography

WDYT?

**Comment by donatasben** [comment]
I totally second that. Would be no need to override the defaults with classes or custom CSS! One step less.

**Reply by aihrig** [reply]
Yes please - really messes with my style that I can't customize this.

**Comment by giltotherescue** [comment]
I'd like this too. Frustrating that I can't control the input focus color.

**Comment by stefanadelbert** [comment]
This is currently tripping me up at the moment too. It's great that the plugin provides some really sane default in a "form reset" way, but it seems crazy to have the power of tailwind for input elements and then not be able to customise something as obvious as the colour with hackery.
Are there any decent workarounds for this?

**Comment by adamwathan** [comment]
Hey all — the way you are supposed to customize the forms plugin is by just writing CSS, for example here's how I'd change the default focus color:

https://play.tailwindcss.com/wJXkrnwcp0?file=css

Don't be afraid to do this sort of thing in your CSS file instead of some special customization DSL — selecting elements and styling them is what CSS is good at, it doesn't make sense for us to reinvest the wheel there.

**Reply by stefanadelbert** [reply]
:pray: @adamwathan 

Thanks for your example for changing the defaults. I confirm that it's working.

Is there a way for me to refence a tw colour, i.e.

```css
@layer base {
  [type="text"]:focus {
    --tw-ring-color: red-400;
  }
}
```

**Reply by stefanadelbert** [reply]
Of course there is... See [theme](https://tailwindcss.com/docs/functions-and-directives#theme)

e.g.

```css
@layer base {
  [type="text"]:focus {
    --tw-ring-color: theme(colors.red.400);
  }
}
```
